### PR TITLE
Start replacer service in server image

### PR DESCRIPTION
**Pull request by rvantonder**
Tries to address #6711. I don't actually know what changes are needed to run replacer in the Docker built image. Putting this up for visibility. Feel free to commandeer this fix/PR.

Test plan: None, WIP. Should absolutely get an e2e test. I think @eseliger is working on one in #6678.

### CODE REVIEW

**Comment by uwedeportivo** [comment]
this will run it in docker single server.

i see you already have it in dev/Procfile for the dev environment (dev/launch.sh and friends)

**Comment by rvantonder** [comment]
> i see you already have it in dev/Procfile for the dev environment (dev/launch.sh and friends)

Indeed, before it a shipping as part of a feature...


Does this: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/cmd/server/shared/globals.go#L12:5


and this:

https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/dev/src-prof-services.json

or this:

https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/enterprise/dev/src-prof-services.json

Need updating?

**Comment by uwedeportivo** [comment]
those two json files and server/shared/globals.go: SrcProfServices are connected (i don't remember anymore how), one is the default of the other. do what other services do, so i guess, yes, add replacer to it

(this part is not a blocker by any means: it's just exposing the /varz, /healthz, /metrics endpoints to the reverse proxy)

**Comment by rvantonder** [comment]
Thanks for the quick review. Looking at [discussion](https://sourcegraph.slack.com/archives/C0EPTDE9L/p1573779154085100?thread_ts=1573778737.083100&cid=C0EPTDE9L) this is not release blocking and I'd like @sourcegraph/automation to have a look and/or merge this along with an e2e test.

**Comment by mrnugget** [comment]
@eseliger is working on an E2E test for automation. And he even had one running, but I'm assuming that that was local.

Merging this now, because if there's a chance, I'd like to get this into this release.

**Comment by slimsag** [comment]
@mrnugget the release has already been cut, nothing more can make it in. You would need to cut a 3.10.1 patch release.

**Comment by mrnugget** [comment]
> @mrnugget the release has already been cut, nothing more can make it in. You would need to cut a 3.10.1 patch release.

Alright! I should check Slack/calendar before emails/PRs 😃

So it looks like this PR is missing the second entry mentioned by @slimsag here? https://github.com/sourcegraph/sourcegraph/issues/6711#issuecomment-555835938

**Comment by slimsag** [comment]
`replacer` is a duplicate of `searcher` effectively, so an entry is missing in the following locations:

- [x] [cmd/server/shared/globals.go](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/cmd/server/shared/globals.go#L15)
- [ ] [cmd/server/shared/shared.go](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/cmd/server/shared/shared.go#L138)
- [ ] [dev/src-prof-services.json](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/dev/src-prof-services.json#L4)
- [ ] [enterprise/dev/src-prof-services.json](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/enterprise/dev/src-prof-services.json#L4)
- [ ] [dev/prometheus/all/prometheus_targets.yml](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/dev/prometheus/all/prometheus_targets.yml#L12-16)
- [ ] [dev/prometheus/linux/prometheus_targets.yml](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@master/-/blob/dev/prometheus/linux/prometheus_targets.yml#L12-16)
- [ ] [deploy-sourcegraph-docker/prometheus/prometheus_targets.yml](https://github.com/sourcegraph/deploy-sourcegraph-docker/blob/master/prometheus/prometheus_targets.yml#L22)

**Comment by mrnugget** [comment]
I've pushed a commit that adds the debug port in the locations mentioned by @slimsag and opened a PR in `deploy-sourcegraph-docker`: https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/52

File: dev/prometheus/all/prometheus_targets.yml [hunk 1]
```diff
@@ -49,4 +49,8 @@
   targets:
     # postgres exporter
     - host.docker.internal:9187
-
+- labels:
+    job: replacer
+  targets:
+    # replacer
+    - host.docker.internal:6076
```
Comment (eseliger) [inline dev/prometheus/all/prometheus_targets.yml]: I think host.docker.internal is not available on docker for Linux 🤔

Comment (mrnugget) [reply dev/prometheus/all/prometheus_targets.yml]: Good catch, although it's the wrong file ;)

**Pull request merged**